### ISSUE TITLE
Arrow keys stop working in the Results panel when at edge of viewable result row

### ISSUE THREAD

**Issue by MikeMKH** [Microsoft/sqlopsstudio#2112]
Issue Type: <b>Bug</b>

Actual: In the Results panel when scrollbar is display use left or right arrow key to navigate to the end of the viewable result row, the cursor will become stuck on the last viewable result cell.

Expected: To be able to navigate freely through the results using the arrow keys.

SQL Operations Studio version: sqlops 0.31.4 (e026ab85a72c190cc4ccc834503aa9fe36dd6407, 2018-07-18T02:14:21.826Z)
OS version: Windows_NT x64 6.1.7601

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-4310M CPU @ 2.70GHz (4 x 2693)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: disabled_software<br>video_decode: enabled<br>video_encode: enabled<br>vpx_decode: unavailable_software<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|7.91GB (2.46GB free)|
|Process Argv|C:\Program Files\SQL Operations Studio\sqlops.exe|
|Screen Reader|no|
|VM|0%|

</details>Extensions: none

**Comment by kburtram** [comment]
@MikeMKH I wasn't able to repro this issue.  Are there any errors in the Help->Developer Tools Console?  Or other repro details that could help locate the problem?

**Comment by BobsCodeEmporium** [comment]
@kburtram  I'm experiencing the same problem as the OP. 

Repro steps are: Select more columns that can fit on the results pane and when you scroll through the results using the cursor keys, it stops at the edge of the visible columns and doesn't want to move anywhere else. The only way to see the rest of the results is to drag the scrollbars with the mouse.

It seems like the "non-visible" columns are out of scope and cannot contain focus, but I'm guessing here.

I can see this in developer tools->console, don't know if it is related:
```
TypeError: Cannot read property 'focus' of undefined
    at setFocus (C:\sqlops-windows\resources\app\node_modules\slickgrid\slick.grid.js:3148)
    at HTMLDivElement.handleClick (C:\sqlops-windows\resources\app\node_modules\slickgrid\slick.grid.js:2964)
    at HTMLDivElement.dispatch (C:\sqlops-windows\resources\app\node_modules\jquery\dist\jquery.js:5110)
    at HTMLDivElement.$event.dispatch (jquery.event.drag-2.3.0.js:382)
    at HTMLDivElement.jQuery.event.add.elemData.handle (C:\sqlops-windows\resources\app\node_modules\jquery\dist\jquery.js:4918)_
```


Version 0.31.4
Commit e026ab85a72c190cc4ccc834503aa9fe36dd6407
Date 2018-07-18T02:14:21.826Z
Shell 1.7.12
Renderer 58.0.3029.110
Node 7.9.0
Architecture x64

**Comment by nickolay** [comment]
I can reproduce, but only when I don't have the vertical scrollbar, e.g. using this query:

    select 'this a long string','this a long string','this a long string','this a long string','this a long string','this a long string','this a long string','this a long string','this a long string','this a long string','this a long string','this a long string','this a long string','this a long string','this a long string','this a long string','this a long string','this a long string'

Hit F5, then click inside the left-most cell, and hit Right-arrow slowly until it reaches the right-most column.

If I increase the number of rows in the result (via `union all select ...`) so that the vertical scrollbar appears, scrolling horizontally via arrow-keys suddenly works. But if I continue fiddling (i.e. scrolling with the mouse / keyboard, clicking inside and outside the Results grid), it eventually breaks.

(same system info as in the previous comment.)

**Comment by nickolay** [comment]
Works for me now in 0.32.7!

**Comment by MikeMKH** [comment]
This has been fixed in 0.32.8
